controlplane/kubeadm/internal/cluster.go

```diff
@@ -135,30 +136,27 @@ func (m *ManagementCluster) GetEtcdCerts(ctx context.Context, cluster types.Name
 	if err := m.Client.Get(ctx, etcdCAObjectKey, etcdCASecret); err != nil {
 		return nil, nil, errors.Wrapf(err, "failed to get secret; etcd CA bundle %s/%s", etcdCAObjectKey.Namespace, etcdCAObjectKey.Name)
 	}
-	etcdCACertData, ok := etcdCASecret.Data[secret.TLSCrtDataName]
+	crtData, ok := etcdCASecret.Data[secret.TLSCrtDataName]
 	if !ok {
-		return nil, nil, errors.New("empty ca certificate")
+		return nil, nil, errors.Errorf("etcd tls crt does not exist for cluster %s/%s", cluster.Namespace, cluster.Name)
 	}
-	etcdCAKeyData, ok := etcdCASecret.Data[secret.TLSKeyDataName]
+	keyData, ok := etcdCASecret.Data[secret.TLSKeyDataName]
 	if !ok {
-		return nil, nil, errors.New("empty ca key")
+		return nil, nil, errors.Errorf("etcd tls key does not exist for cluster %s/%s", cluster.Namespace, cluster.Name)
 	}
-	return etcdCACertData, etcdCAKeyData, nil
+	return crtData, keyData, nil
 }
 
-// TargetClusterEtcdIsHealthy runs a series of checks over a target cluster's etcd cluster.
-// In addition, it verifies that there are the same number of etcd members as control plane Machines.
-func (m *ManagementCluster) TargetClusterEtcdIsHealthy(ctx context.Context, clusterKey types.NamespacedName, controlPlaneName string) error {
-	cluster, err := m.getCluster(ctx, clusterKey)
-	if err != nil {
-		return err
-	}
-	resp, err := cluster.etcdIsHealthy(ctx)
+type healthCheck func(context.Context) (healthCheckResult, error)
+
+// healthCheck runs the check and does various error manipulation that is the sam
+func (m *ManagementCluster) healthCheck(ctx context.Context, check healthCheck, clusterKey types.NamespacedName, controlPlaneName string) error {
```

Comment:
Should we consider allowing this to perform multiple `healthCheck`s? Then we could expose it directly and let the consumer choose if they want to test etcd health, control plane health, both, or even add some additional check we may need in the future.
```suggestion
func (m *ManagementCluster) CheckHealth(ctx context.Context, clusterKey types.NamespacedName, controlPlaneName string, checks ...healthCheck) error {
```

Reply:
The way that ManagementCluster gets the target cluste client makes this particular change need a larger refactoring the the suggestion. I think the explicit list of calls is ok for now as it will be quite readable from the reconciler, but if we decide that's not the case going forward we can revisit in the future.